cli/ops.rs

```diff
@@ -176,6 +176,48 @@ pub fn dispatch_all_legacy(
       );
       Op::Async(result_fut)
     }
+    Ok(Op::AsyncOptional(fut)) => {
```

Comment:
Can you explain? I don't get it.

Reply:
It actually act almost the same as `Op::Async` for this part, only that the result wrapping enum is different. I'll see a way if I could avoid duplicate code here

Reply:
What's the purpose of this extra variant?
To make sure the process will exit even when the future is not ready?
I'd suggest to use just Op::Async() and have a separate pending-op count rather than using `pending_ops.len()`.

Reply:
Maybe I'm misunderstanding, but if both are marked as `Op::Async` I don't think `Isolate` will be able to tell if a future is optional when polling?

I kind of want to avoid manipulating `Isolate` directly from cli (and I'm not quite sure if changing return types of op selectors is a good idea). Also not sure about the counter part since no differentiation between things like enum types mean there is no way I could just decrement for required futures, unless we want to chain another step after the user provided future -- and thus also need to make the counter atomic

Reply:
We need something like uv_unref ... I’m going to punt on this problem until next week - not sure what the right way to handle this is.